roles/libvirt_rhel_vm/molecule/shared/prepare.yml

```diff
@@ -0,0 +1,44 @@
+- name: register RHSM
+  hosts: all
+  pre_tasks:
+    - name: install python-apt where necessary
+      pip:
+        # yamllint disable-line rule:line-length
+        name: git+https://git.launchpad.net/ubuntu/+source/python-apt@applied/ubuntu/bionic-updates
+        state: present
+      when: ansible_pkg_mgr == "apt"
+
+    - name: download image file
+      get_url:
+        dest: "~/"
+        # yamllint disable-line rule:line-length
+        url: https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2.xz
+
+    - name: extract file
+      command: unxz CentOS-7-x86_64-GenericCloud.qcow2.xz
+      args:
+        chdir: "~/"
+        creates: "~/CentOS-7-x86_64-GenericCloud.qcow2"
+
+    - name: create nic-config directory
```

Comment:
Could the [_tempfile_](https://docs.ansible.com/ansible/latest/modules/tempfile_module.html) module be used here?

Reply:
Negative. This is a test dir that needs to persist into the next `playbook.yml` run. The info of the location of the tempdir will be lost after `prepare.yml` is executed.

Reply:
For the cases where it looks like a module could be used but isn't, I would add a comment explaining why.

Reply:
Comment added.